gplugins/design_recipe/DesignRecipe.py

```diff
@@ -84,9 +154,332 @@ def eval_dependencies(self, force_rerun_all=False) -> bool:
         Evaluate this `DesignRecipe`'s dependencies.
         Because `dependencies` are assumed to be independent,
         they can be evaluated in any order.
+
+        Parameters:
+            force_rerun_all: Forces design recipes to be re-evaluated
         """
         success = True
         for recipe in self.dependencies:
-            if force_rerun_all or (not recipe.is_fresh):
+            if force_rerun_all or (not recipe.is_fresh()):
                 success = success and recipe.eval(force_rerun_all)
         return success
+
+    def load_recipe_results(self):
+        """
+        Loads recipe results from pickle file into class attribute
+        """
+        self.recipe_results = self.recipe_results.get_pickle()
+
+    def save_recipe_results(self):
+        """
+        Saves recipe_results to pickle file while adding setup information.
+        This includes:
+        - component hash
+        - layerstack
+        - convergence_settings
+        - simulation_settings
+
+        This is usually done after convergence testing is completed and simulation
+        settings are accurate and should be saved for future reference/recall.
+        """
+        self.recipe_results.recipe_setup = self.recipe_setup
+        self.recipe_dirpath.resolve().mkdir(parents=True, exist_ok=True)
+        self.recipe_results.save_pickle()
+
+    def is_same_recipe_results(self) -> bool:
+        """
+        Returns whether recipe results' setup are the same as the current setup
+        for the recipe. This is important for preventing hash collisions.
```

Comment:
wym by 'preventing hash collisions' here? is there some ambiguity that's resolved by this flag (e.g. is the hash different because the simulation settings are different)?

Reply:
Different recipe setups may create the same hash since the hashes are fixed length. To avoid this collision, double check the pickled setup with the current setup.

Reply:
I don't know if we are worried about hash collisions- this feels kind of hacky, but you can leave it if you'd prefer